lab_2/main.py

```diff
@@ -34,7 +55,36 @@ def fill_lcs_matrix(first_sentence_tokens: tuple, second_sentence_tokens: tuple)
     :param second_sentence_tokens: a tuple of tokens
     :return: a lcs matrix
     """
-    pass
+    is_tuples_correct = isinstance(first_sentence_tokens, tuple) and isinstance(second_sentence_tokens, tuple)
+    is_both_tuples_empty = not first_sentence_tokens or not second_sentence_tokens
+    if not is_tuples_correct or is_both_tuples_empty:
+        return []
+    for word_1s in first_sentence_tokens:
+        if not word_1s and not isinstance(word_1s, str):
+            return []
+    for word_2s in second_sentence_tokens:
+        if not word_2s and not isinstance(word_2s, str):
+            return []
+
+    rows_first_sent = len(first_sentence_tokens)
+    columns_second_sent = len(second_sentence_tokens)
+
+    lcs_matrix = create_zero_matrix(rows_first_sent + 1, columns_second_sent + 1)
+
+    for first_sent_word in range(1, rows_first_sent + 1):
+        for second_sent_word in range(1, columns_second_sent + 1):
+            if first_sentence_tokens[first_sent_word - 1] == second_sentence_tokens[second_sent_word - 1]:
+                lcs_matrix[first_sent_word][second_sent_word] = lcs_matrix[first_sent_word - 1][second_sent_word - 1] \
+                                                                + 1
+            else:
+                lcs_matrix[first_sent_word][second_sent_word] = max(lcs_matrix[first_sent_word - 1][second_sent_word],
+                                                                    lcs_matrix[first_sent_word][second_sent_word - 1])
+
+    del lcs_matrix[0]
+    for row in lcs_matrix:
+        del row[0]
```

Comment:
Please, leave some explanation.

Reply:
In this function zero matrix is created with one row and one column more than lengths of sentences: we need something to substract from when filling the matrix for first and last element of each row (as there is no number). After this, zero row and zero column are being removed, so filled matrix corresponds lengths of sentences.

Reply:
Could not it be done without adding empty column and row?

Reply:
I can think only of if/elif/else and checking if row or column are first (with index = 0)